pkg/streamingpromql/operators/function_over_instant_vector_test.go

```diff
@@ -62,3 +63,68 @@ func TestFunctionOverInstantVector(t *testing.T) {
 	require.True(t, metadataFuncCalled, "Supplied MetadataFunc must be called matching the signature")
 	require.Equal(t, len(inner.data), seriesDataFuncCalledTimes, "Supplied SeriesDataFunc was called once for each Series")
 }
+
+func TestFunctionOverInstantVectorWithScalarArgs(t *testing.T) {
+	inner := &testOperator{
+		series: []labels.Labels{
+			labels.FromStrings("series", "0"),
+			labels.FromStrings("series", "1"),
+		},
+		data: []types.InstantVectorSeriesData{
+			{Floats: []promql.FPoint{{T: 0, F: 1}}},
+			{Floats: []promql.FPoint{{T: 0, F: 2}}},
+		},
+	}
+
+	scalarOperator1 := &testScalarOperator{
+		value: types.ScalarData{Samples: []promql.FPoint{{T: 0, F: 3}}},
+	}
+
+	scalarOperator2 := &testScalarOperator{
+		value: types.ScalarData{Samples: []promql.FPoint{{T: 60, F: 4}}},
+	}
+
+	seriesDataFuncCalledTimes := 0
+	mustBeCalledSeriesData := func(_ types.InstantVectorSeriesData, scalarArgs []types.ScalarData, _ *limiting.MemoryConsumptionTracker) (types.InstantVectorSeriesData, error) {
+		seriesDataFuncCalledTimes++
+		// Verify that the scalar arguments are correctly passed and in the order we expect
+		require.Equal(t, 2, len(scalarArgs))
+		require.Equal(t, 3.0, scalarArgs[0].Samples[0].F)
+		require.Equal(t, 4.0, scalarArgs[1].Samples[0].F)
```

Comment:
This could be modified to something like this, which will also check everything else about the scalar data is as we expect:

```suggestion
		require.Equal(t, 2, len(scalarArgs))
		require.Equal(t, types.ScalarData{Samples: []promql.FPoint{{T: 0, F: 3}}}, scalarArgs[0])
		require.Equal(t, types.ScalarData{Samples: []promql.FPoint{{T: 60, F: 4}}}, scalarArgs[1])
```